src/freenas/usr/local/bin/initrd-zfs.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+
+import libzfs
+import pyudev
+
+
+if __name__ == "__main__":
+    boot_pool, root, update_initramfs_if_changes = sys.argv[1:]
+
+    with libzfs.ZFS() as zfs:
+        disks = [disk.replace("/dev/", "") for disk in zfs.get(boot_pool).disks]
+
+    mapping = {}
+    for dev in filter(
+        lambda d: not d.sys_name.startswith("sr") and d.get("DEVTYPE") in ("disk", "partition"),
+        pyudev.Context().list_devices(subsystem="block")
+    ):
+        if dev.get("DEVTYPE") == "disk":
+            mapping[dev.sys_name] = dev.get("ID_BUS")
+        elif dev.get("ID_PART_ENTRY_UUID"):
+            parent = dev.find_parent("block")
+            mapping[dev.sys_name] = parent.get("ID_BUS")
+            mapping[os.path.join("disk/by-partuuid", dev.get("ID_PART_ENTRY_UUID"))] = parent.get("ID_BUS")
+
+    has_usb = False
+    for dev in disks:
+        if mapping.get(dev) == "usb":
+            has_usb = True
+
+    zfs_config_path = os.path.join(root, "etc/default/zfs")
+    with open(zfs_config_path) as f:
+        original_config = f.read()
+        lines = original_config.rstrip().split("\n")
+
+    zfs_var_name = "ZFS_INITRD_POST_MODPROBE_SLEEP"
+    lines = [line for line in lines if not line.startswith(f"{zfs_var_name}=")]
+    if has_usb:
+        lines.append(f"{zfs_var_name}=15")
+
+    with open(zfs_config_path, "w") as f:
+        new_config = "\n".join(lines) + "\n"
+        f.write(new_config)
+
+    if update_initramfs_if_changes == "1" and new_config != original_config:
```

Comment:
Should we write zfs_config_path only here? so we dont write the config if nothing has changed?